File: miniapp/src/api/issue.ts
import { request } from '@/utils/request'
import { BASE_URL } from '@/utils/constants'

export interface Issue {
  id: number
  bizCode: string
  storeId: string
  storeName: string
  title: string
  issueType: string
  subType: string
  urgency: string          // urgent | normal | low
  description: string
  xiangmuId: string
  syncStatus: string
  status: string           // pending | processing | pending_acceptance | resolved | closed
  recordsData: string
  replyText: string        // 解决原因
  processedBy: string
  createdAt: string
  updatedAt: string
}

export interface IssuePageResp {
  records: Issue[]
  total: number
  current: number
  size: number
}

export interface IssueOverview {
  processing: number
  pendingAcceptance: number
  resolved: number
  all: number
}

export function fetchIssueList(params?: { status?: string; pageNum?: number; pageSize?: number; all?: boolean }) {
  return request<IssuePageResp>({ url: '/issue/list', data: params, showLoading: false })
}

export function reportIssueClick() {
  return request({ url: '/issue/report-click', method: 'POST', showLoading: false, silent: true })
}

export function fetchIssueOverview(all?: boolean) {
  return request<IssueOverview>({ url: '/issue/overview', data: all ? { all: true } : undefined, showLoading: false, silent: true })
}

export function fetchIssueOverviewStores() {
  return request<any[]>({ url: '/issue/overview-stores', showLoading: false, silent: true })
}

export function fetchIssueDetail(id: number) {
  return request<Issue>({ url: `/issue/${id}`, showLoading: false })
}

export function createIssue(data: {
  title: string
  issueType: string
  subType?: string
  urgency: string
  description: string
  images?: string
  contactName: string
  contactPhone: string
}) {
  return request<Issue>({ url: '/issue', method: 'POST', data })
}

export function acceptIssue(id: number, remark?: string) {
  return request<Issue>({ url: `/issue/${id}/accept`, method: 'POST', data: { remark: remark || '' } })
}

/** 同步外部表单提交结果 */
export function syncExternalIssue(data: { id: number; issueNo?: string; status?: string }) {
  return request<Issue>({ url: '/issue/sync-external', method: 'POST', data })
}

/** 按外部 ID 查 issue */
export function fetchIssueByExternalId(externalId: number) {
  return request<Issue>({ url: `/issue/by-external/${externalId}`, showLoading: false })
}

/** 代理查询外部处理记录 */
export function fetchIssueRecords(id: number) {
  return request<any>({ url: `/issue/${id}/records`, showLoading: false })
}

/** 门店确认（已解决/未解决） */
export function storeConfirm(id: number, action: 'accept' | 'reject') {
  return request<any>({ url: `/issue/${id}/store-confirm?action=${action}`, method: 'POST' })
}

/** 代理门店回复 */
export function replyIssue(id: number, replyText: string, mediaUrls?: string) {
  return request<any>({ url: `/issue/${id}/reply`, method: 'POST', data: { replyText, mediaUrls: mediaUrls || '' } })
}

/** 代理上传文件到 task_platform */
export function uploadIssueMedia(filePaths: string[], chatId?: string) {
  return new Promise<any>((resolve, reject) => {
    const baseUrl = (BASE_URL as string).replace(/\/api\/mp\/?$/, '')
    uni.uploadFile({
      url: `${baseUrl}/api/mp/issue/upload-media`,
      filePath: filePaths[0],
      name: 'files',
      formData: { chatId: chatId || '' },
      header: { Authorization: `Bearer ${uni.getStorageSync('token')}` },
      success: (res: any) => {
        try { const d = JSON.parse(res.data); if (d.code === 200) resolve(d.data); else reject(d) }
        catch { reject(res) }
      },
      fail: reject,
    })
  })
}

/** 上传现场图片，复用通用凭证上传接口，返回图片 URL */
export function uploadIssueImage(filePath: string) {
  return new Promise<string>((resolve, reject) => {
    const baseUrl = BASE_URL.replace(/\/api\/mp\/?$/, '')
    uni.uploadFile({
      url: `${baseUrl}/api/mp/upload/voucher`,
      filePath,
      name: 'file',
      header: { Authorization: `Bearer ${uni.getStorageSync('token')}` },
      success: (res: any) => {
        try { const d = JSON.parse(res.data); if (d.code === 200) resolve(d.data.url); else reject(d) }
        catch { reject(res) }
      },
      fail: reject,
    })
  })
}

File: xzcpc-xzg/mp/src/main/java/com/xzcpc/mp/entity/Issue.java
package com.xzcpc.mp.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * P0 B1: 问题处理单
 *
 * 状态流转（象掌柜侧 5 态）：
 * pending（已提交）→ processing（处理中）→ pending_acceptance（待验收）→ resolved（已解决）
 * closed（已关闭）为终态：合并/无需处理
 *
 * 派单与处理由象目经理外部系统完成，象掌柜仅同步简化状态 + 处理结果。
 */
@Data
@TableName("issue")
public class Issue {
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 业务编码 */
    private String bizCode;

    /** 提交门店ID */
    private String storeId;

    /** 门店名称 */
    private String storeName;

    /** 问题标题 */
    private String title;

    /** 问题类型：设备问题|物料问题|物流问题|系统问题|经营异常|其他 */
    private String issueType;

    /** 子类型（如设备问题下的制冰机/净水器） */
    private String subType;

    /** 紧急程度：urgent|normal|low */
    private String urgency;

    /** 问题描述 */
    private String description;

    /** 象目经理同步单号 */
    private String xiangmuId;

    /** 象目经理同步状态：pending|synced|failed */
    private String syncStatus;

    /** 状态：pending|processing|pending_acceptance|resolved|closed */
    private String status;

    /** 来源渠道：MINI_PROGRAM(小程序)|FEISHU_GROUP(飞书群H5)|HQ(总部上报)|null(其他旧数据) */
    private String source;

    /** 处理记录 JSON（records + replies + solutionPhotos），task_platform 回调推送 */
    private String recordsData;

    /** 解决原因（门店回复的未解决原因文本） */
    @TableField("acceptance_remark")
    private String replyText;

    /** 处理人 */
    private String processedBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableLogic(value = "0", delval = "1")
    private Integer delFlag;
    @Version
    private Integer version;

    @TableField(exist = false)
    private String supervisorName;
}
